src/Visualisation-App.tsx

```diff
@@ -1,32 +1,121 @@
-import Viewer from "./viewer";
+import React from "react";
+import Viewer from "./Viewer";
+import ViewerOptions from "./ViewerOptions";
+import "./Visualisation-App.css";
 import { useState } from "react";
-import { searchResultsHits } from "./searchResultsHits.interface";
-import Searchbar from "./searchbar";
-import SearchResults from "./searchResults";
+import Searchbar from "./Searchbar";
+import SearchResults from "./SearchResults";
+import { PluginContext } from "molstar/lib/mol-plugin/context";
+import { SearchResultsHits } from "./SearchResults";
+import { setSubtreeVisibility } from "molstar/lib/mol-plugin/behavior/static/state";
 
-function VisualisationApp() {
+const VisualisationApp: React.FunctionComponent = () => {
     const [url, setUrl] = useState("");
+    // const url = "https://alphafold.ebi.ac.uk/files/AF-P04350-F1-model_v1.cif";
+    const plugin =
+        React.useRef<PluginContext | null>() as React.MutableRefObject<PluginContext | null>;
+
     const [searchResults, setSearchResults] = useState<
-        searchResultsHits[] | null
+        SearchResultsHits[] | null
     >(null);
 
+    async function addAlphafoldColour() {
+        //if alphafold view already exists
+        for (const c of plugin.current!.managers.structure.hierarchy.current
```

Comment:
Is there a chance that any of `plugin.current!.managers.structure.hierarchy.current` would be null or undefined, I'm scared of big access chains like this (and would probably safeguard with optional chaining.

Reply:
It shouldn't be null but that's a good point. I can try using the ?. operator instead. What is the conventional way to handle a situation in which there are multiple chains which depend each other? For example, this snippet from Viewer.tsx: 
```
const _data = await plugin.current?.builders.data.download(
            { url: url },
            { state: { isGhost: true } }
        );
const trajectory = await plugin.current?.builders.structure.parseTrajectory(
                _data,
                "mmcif"
            );
await plugin.current?.builders.structure.hierarchy.applyPreset(
            trajectory,
            "default"
        );
```
How do I ensure that _data and trajectory exist for the second and third commands. Do I need to wrap them in checks or tries?

Reply:
A lazy way is probably to wrap it in a try, gracefully handle the exception case and report the error, it might not be a problem 🤷, but safer to handle it than have the app just whitescreen.